Source/BoundaryConditions/PEC_Insulator.H

```diff
@@ -0,0 +1,174 @@
+     *        and the normal fields extrapolated from the valid cells.
+     *
+     * \param[in,out] Efield
+     * \param[in]     ng_fieldgather      number of guard cells used by field gather
+     * \param[in]     geom                geometry object of level "lev"
+     * \param[in]     lev                 level of the Multifab
+     * \param[in]     patch_type          coarse or fine
+     * \param[in]     ref_ratios          vector containing the refinement ratios of the refinement levels
+     * \param[in]     time                current time of the simulation
+     * \param[in]     split_pml_field     whether pml the multifab is the regular Efield or
+     *                                    split pml field
+     */
+    void ApplyPEC_InsulatortoEfield (std::array<amrex::MultiFab*, 3> Efield,
+                                     amrex::Array<FieldBoundaryType,AMREX_SPACEDIM> const & field_boundary_lo,
+                                     amrex::Array<FieldBoundaryType,AMREX_SPACEDIM> const & field_boundary_hi,
+                                     amrex::IntVect const & ng_fieldgather, amrex::Geometry const & geom,
+                                     int lev, PatchType patch_type, amrex::Vector<amrex::IntVect> const & ref_ratios,
+                                     amrex::Real time,
+                                     bool split_pml_field = false);
+    /**
+     * \brief Apply either the PEC or insulator boundary condition on the boundary and in the
+     *        guard cells.
+     *        In the PEC, the nodal fields (in a Yee mesh) as made even relative to the boundary,
+     *        the non-nodal fields are made odd.
+     *        In the insulator, the tangential fields are set to the value if specified, otherwise unchanged,
+     *        and the normal fields extrapolated from the valid cells.
+     *
+     * \param[in,out] Bfield
+     * \param[in]     ng_fieldgather      number of guard cells used by field gather
+     * \param[in]     geom                geometry object of level "lev"
+     * \param[in]     lev                 level of the Multifab
+     * \param[in]     patch_type          coarse or fine
+     * \param[in]     ref_ratios          vector containing the refinement ratios of the refinement levels
+     * \param[in]     time                current time of the simulation
+     */
+    void ApplyPEC_InsulatortoBfield (std::array<amrex::MultiFab*, 3> Bfield,
+                                     amrex::Array<FieldBoundaryType,AMREX_SPACEDIM> const & field_boundary_lo,
+                                     amrex::Array<FieldBoundaryType,AMREX_SPACEDIM> const & field_boundary_hi,
+                                     amrex::IntVect const & ng_fieldgather, amrex::Geometry const & geom,
+                                     int lev, PatchType patch_type, amrex::Vector<amrex::IntVect> const & ref_ratios,
+                                     amrex::Real time);
```

Comment:
Why don't we need a `split_pml_field` argument for B, as for E? In other words, why is the splitting of B, which I think occurs in the PML, not relevant in this case?

Reply:
Good question. I copied this pattern from the PEC boundary conditions. Do you know why the split is done there?

Reply:
Ah interesting, I don't know. I think this would be a question for @RevathiJambunathan or @roelof-groenewald.

Reply:
Pinging @RevathiJambunathan and @roelof-groenewald just to make sure that the PR doesn't get stalled.

Reply:
Re-Pinging @RevathiJambunathan and @roelof-groenewald.

Reply:
Sorry for the slow response. It looks like the flag was introduced in https://github.com/ECP-WarpX/WarpX/pull/2541 to fix an observed bug when one boundary is PEC and an adjacent one is PML. @RevathiJambunathan would know better but I expect it is not needed to apply the same to the B-field since the update to the E-field ensures that the B-field values come out appropriately in the B-field push.